src/react/components/SaleorProvider.tsx

```diff
@@ -1,34 +1,23 @@
-import React, { Dispatch, SetStateAction } from "react";
-import { Core, SaleorSDK, SaleorSDKConfig } from "../../core";
+import React from "react";
+import { SaleorSDK } from "../../core";
 import { ApolloClient, NormalizedCacheObject } from "@apollo/client";
+import { SaleorContext, SaleorContextType } from "../context";
 
-export type SaleorContextType = {
-  api: Core;
+export const SaleorProvider: React.FC<{
   client: ApolloClient<NormalizedCacheObject>;
   channel: string;
-  setChannel: Dispatch<SetStateAction<string>>;
-};
-
-export const SaleorContext = React.createContext<SaleorContextType | null>(
-  null
-);
-
-export const SaleorProvider: React.FC<{ config: SaleorSDKConfig }> = ({
-  config: { apiUrl, channel: initChannel },
-  children,
-}) => {
-  const [context, setContext] = React.useState<SaleorContextType | null>(null);
+}> = ({ client, channel: initChannel, children }) => {
```

Comment:
The `client` is actually `createSaleorClient()` instance (or `SaleorSDK` as it's called). Usage:

```
// react app index.tsx

import { createSaleorClient, SaleorProvider } from "@saleor/sdk";

const saleorClient = createSaleorClient({ channel: "default-channel", apiUrl: "https://instance.saleor.cloud/graphl/" })

const App = () => (
  <SaleorProvider client={saleorClient}>
    ...
  </SaleorProvider>
)
```